src/util/operations.spec.tsx

```diff
@@ -0,0 +1,37 @@
+import { getInstanceName } from "./operations";
+import { LxdOperation } from "types/operation";
+
+const craftOperation = (url: string) => {
+  return {
+    resources: {
+      instances: [url],
+    },
+  } as LxdOperation;
+};
+
+describe("getInstanceName", () => {
+  it("instance operation in custom project", () => {
+    const operation = craftOperation("/1.0/instances/testInstance1");
+    const name = getInstanceName(operation);
+
+    expect(name).toBe("testInstance1");
+  });
+
+  it("instance operation in custom project", () => {
+    const operation = craftOperation(
+      "/1.0/instances/testInstance2?project=project"
+    );
+    const name = getInstanceName(operation);
+
+    expect(name).toBe("testInstance2");
+  });
+
+  it("snapshot operation", () => {
+    const operation = craftOperation(
+      "/1.0/instances/testInstance3/snapshots/testSnap"
+    );
+    const name = getInstanceName(operation);
+
+    expect(name).toBe("testInstance3");
+  });
```

Comment:
Maybe we can branch here as well, and check the snapshot operation both for the default and for a custom project?